### PR TITLE
Add cleaner for volumes

### PR DESCRIPTION
**Pull request by erkanerol**
Towards https://github.com/giantswarm/giantswarm/issues/25787

We don't have a cleaner for volumes (named disks) right now. We will inject ClusterID to NamedDisks as description with this PR https://github.com/vmware/cloud-director-named-disk-csi-driver/pull/174.

This PR adds a cleaner for volumes. The cleaner detaches attached VMs and deletes disks.  

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/vcd/volume.go
Comment (kopiczko) [inline pkg/vcd/volume.go:14]: Should be `ClusterId` -> `clusterId` I think

**Review by kopiczko — APPROVED** [review]
LGTM if they accept your PR upstream

**Review by jkremser — APPROVED** [review]
👍

File: pkg/vcd/volume.go
Comment (jkremser) [inline pkg/vcd/volume.go:15]: has this always been there? I've checked the records using the cd-cli and it's really there for some disks. Was I blind or you've added some smartness also somewhere else (CSI maybe)?

```
  {
    "Xmlns": "",
    "HREF": "https://vmware.ikoula.com/api/disk/93caa223-c315-4994-82bd-02e8e883ac26",
    "Id": "",
    "Type": "",
    "Name": "pvc-3e1ec10b-fcaa-4b16-97ab-693dcf0ed471",
    "Vdc": "https://vmware.ikoula.com/api/vdc/2ad2288b-f5fd-4410-bbc5-9e470bf4ae2e",
    "SizeMb": 1024,
    "Iops": 0,
    "Encrypted": false,
    "UUID": "6000c294-521a-7d6f-4cc4-18ab46456330",
    "DataStore": "",
    "DataStoreName": "",
    "OwnerName": "xaviertn",
    "VdcName": "vDC 73640",
    "Task": "",
    "StorageProfile": "https://vmware.ikoula.com/api/vdcStorageProfile/69a217fb-7f8a-45fb-aea9-860bbcff9131",
    "StorageProfileName": "vSAN-RAID-5",
    "Status": "RESOLVED",
    "BusType": "6",
    "BusSubType": "pvscsi",
    "BusTypeDesc": "Paravirtual (SCSI)",
    "AttachedVmCount": 0,
    "SharingType": "None",
    "IsAttached": false,
    "IsShareable": false,
    "Description": "NO_RDE_4f0dcd2a-18b6-4ccc-9870-6af0d53381e1",
    "Link": null,
    "Metadata": null
  }
```

Comment (jkremser) [reply pkg/vcd/volume.go:15]: meh, scratch that ^ I didn't read the PR description properly :D

File: pkg/vcd/volume.go
Comment (erkanerol) [reply pkg/vcd/volume.go:18]: Yes. I did many tests by customizing pageSize and generating dummy PVs.

**Comment by erkanerol** [comment]
My upstream PR is approved by two maintainers. I assume it is going to be merged.

**Pull request merged**
